library/Scientist/Duration.hs

```diff
@@ -1,21 +1,68 @@
 {-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 
 module Scientist.Duration
-  ( Duration(..)
+  ( Duration
   , measureDuration
+  , toSecs
+  , fromSecs
+  , toNanoSecs
+  , fromNanoSecs
```

Comment:
Personally when I was first attempting to utilize this type `unDuration` and the lack of numeric instances was confusing. The API that now exists in this PR feels more complete and understandable to me.

Reply:
Calling it `DurationSeconds` would've probably helped.Giving it [`ToMetricValue]`(https://hackage.haskell.org/package/datadog-0.3.0.0/docs/Network-StatsD-Datadog.html#t:ToMetricValue) (and perhaps `ToJSON`) may be the way to have an absolutely minimal but useful/understandable interface.

Reply:
Allowing these to be manipulated and constructed like regular numbers is a net positive to me. I have no idea what someone will want to do in their `publish` function, so giving them the primitives affords flexibility.

Reply:
That's fair. I'm still not blocking, but how about exposing a `durationToSeconds` and numeric instances only?

I think the other functions are unnecessary. You make a good argument for completeness and treating it like a number, but I don't think it qualifies for the otherwise-unused `from/toNanoSecs` and `fromSecs` functions. And I think the name `durationToSeconds` can help on your understandability point.

WDYT?